app/view/sdl/ServiceUpdatePopUp.js

```diff
@@ -0,0 +1,184 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+/**
+ * @name SDL.ServiceUpdatePopUp
+ * @desc Service update pop up module visual representation
+ * @category View
+ * @filesource app/view/sdl/ServiceUpdatePopUp.js
+ * @version 1.0
+ */
+
+SDL.ServiceUpdatePopUp = Em.ContainerView.create(
+  {
+    elementId: 'ServiceUpdatePopUp',
+    classNames: 'ServiceUpdatePopUp',
+    classNameBindings: [
+      'active'
+    ],
+    childViews: [
+      'message',
+      'progressIndicatorView',
+      'backButton'
+    ],
+
+    timer: null,
+    active: false,
+    content: '',
+    progress: false,
+
+    progressIndicatorView: Em.View.extend(
+      {
+        elementId: 'progress',
+        classNameBindings: 'this.parentView.progress:progress'
+      }
+    ),
+    backButton: SDL.Button.extend(
+      {
+        classNames: 'button backButton',
+        text: 'X',
+        click: function() {
+          this._parentView.deactivate();
+        },
+        buttonAction: true,
+        onDown: false
+      }
+    ),
+    message: SDL.Label.extend(
+      {
+        elementId: 'text',
+        classNames: 'text',
+        contentBinding: 'parentView.content'
+      }
+    ),
+
+    /**
+     * @function deactivate
+     * @desc Deactivate ServiceUpdatePopUp
+     * 
+     */
+    deactivate: function() {
+      this.set('active', false);
+      this.set('progress', false);
+      this.set('content', '');
+    },
+
+    /**
+     * @function activate
+     * @desc activate ServiceUpdatePopUp
+     * @param {string} serviceType 
+     * @param {string} serviceEvent 
+     * @param {string} serviceReason 
+     */
+    activate: function(serviceType, serviceEvent, serviceReason) {
+      if(serviceType === undefined) {
+        return;
+      }
+
+      switch (serviceEvent) {
+        case 'REQUEST_RECEIVED': {
+          if(serviceType == 'VIDEO') {
+            this.set('content', 'Starting Video Stream');
+          }
+          else if(serviceType == 'AUDIO') {
+            this.set('content', 'Starting Audio Stream');
+          }
+          else {
+            return;
+          }
+          this.set('progress', true);
+          this.set('active', true);
+          break;
+        }
+        case 'REQUEST_ACCEPTED': {
+          this.deactivate();
+          break;
+        }
+        case 'REQUEST_REJECTED': {
+          this.rejectedRequest(serviceReason);
+          var self = this;
+          this.timer = setTimeout(
+            function() {
+              self.deactivate();
+            },
+            5000
+          );
+        }
+        default: {
+          if(serviceReason !== undefined) {
```

Comment:
@ValeriiMalkov I think that check of this condition is redundant.

Reply:
@BSolonenko We can receive the message without the `serviceEvent` parameter, but with `serviceReason` parameter or without.  In case `serviceReason` comes without `serviceEvent` it means that request was rejected. In other case when we don't receive both parameter, we should ignore this message. That's why we should check this condition. This is not redundant.